embedded-graphics/src/image/mod.rs

```diff
@@ -1,33 +1,262 @@
+//! let mut display: Display<Rgb565> = Display::default();
+//!
+//! let tga =
+//!     Tga::from_slice(include_bytes!("../../../simulator/examples/rust-pride.tga")).unwrap();
+//!
+//! let image = Image::new(&tga, Point::zero());
+//!
+//! image.draw(&mut display);
+//!
+//! # Ok::<(), core::convert::Infallible>(())
+//! ```
+//!
+//! [tinytga]: https://crates.io/crates/tinytga
+//! [`IntoPixelIter`]: ./trait.IntoPixelIter.html
+//! [`ImageDimensions`]: ./trait.ImageDimensions.html
+//! [`Image`]: ./struct.Image.html
+//! [`DrawTarget`]: ../trait.DrawTarget.html
 
-#[cfg(feature = "bmp")]
-mod image_bmp;
 mod image_raw;
-#[cfg(feature = "tga")]
-mod image_tga;
 
-pub use self::image_raw::{Image, ImageBE, ImageLE};
+pub use self::image_raw::{ImageRaw, ImageRawBE, ImageRawLE};
 
-#[cfg(feature = "bmp")]
-pub use self::image_bmp::ImageBmp;
-#[cfg(feature = "tga")]
-pub use self::image_tga::ImageTga;
+use crate::{
+    draw_target::DrawTarget,
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    transform::Transform,
+};
+use core::{fmt, marker::PhantomData};
 
-/// Image file trait.
-pub trait ImageFile<'a>: crate::geometry::Dimensions + Sized {
-    /// Create a new image with given input file
-    ///
-    /// The input file is expected to be of a particular format (BMP, TGA, etc) and contain file
-    /// metadata like width/height and pixel data. Because parsing may fail, this returns a
-    /// `Result<Self, ()>`.
-    fn new(filedata: &'a [u8]) -> Result<Self, ()>;
+/// Image data iterator trait
+pub trait IntoPixelIter<C>
+where
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Iterator over pixels in the image
+    type PixelIterator: Iterator<Item = Pixel<C>>;
+
+    /// Get an iterator over the pixels of the image
+    fn pixel_iter(self) -> Self::PixelIterator;
+}
 
+/// Image data trait
+///
+/// This trait provides an interface to common image metadata. It should be implemented along with
+/// [`IntoPixelIter`] for full embedded-graphics integration.
+pub trait ImageDimensions {
     /// Get the width in pixels of an image
     fn width(&self) -> u32;
 
     /// Get the height in pixels of an image
     fn height(&self) -> u32;
 }
+
+/// Image support for embedded-graphics
+///
+/// The `Image` struct serves as a wrapper around other image types that provide pixel data decoded
+/// from a given format (raw bytes, BMP, TGA, etc).
+///
+/// `Image` accepts any item that implements `ImageDimensions` and `&'_ IntoPixelIter`.
+///
+/// Refer to the [module documentation] for examples.
+///
+/// [module documentation]: ./index.html
+#[derive(Debug, Clone, Copy)]
+pub struct Image<'a, I, C> {
+    image_data: &'a I,
+    offset: Point,
+    c: PhantomData<C>,
+}
+
+impl<'a, I, C> Image<'a, I, C>
+where
+    &'a I: IntoPixelIter<C>,
+    I: ImageDimensions,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Create a new `Image` with a given [`IntoPixelIter`]
```

Comment:
The parameter doesn't only need to implement `IntoPixelIter` but also the `ImageDimensions` trait. So maybe the docs could use a more generic way of referring to the parameter by using `... the given image data` instead. An additional description of the required trait impls can then be added in the next paragraph.